src/main/java/com/puppycrawl/tools/checkstyle/site/ClassAndPropertiesSettersJavadocScraper.java

```diff
@@ -0,0 +1,168 @@
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2023 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.site;
+
+import java.beans.Introspector;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.puppycrawl.tools.checkstyle.FileStatefulCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+import com.puppycrawl.tools.checkstyle.utils.BlockCommentPosition;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
+
+/**
+ * Class for scraping class javadoc and all property setter javadocs from the
+ * given checkstyle module.
+ */
+@FileStatefulCheck
+public class ClassAndPropertiesSettersJavadocScraper extends AbstractJavadocCheck {
+
+    /**
+     * Map of scraped javadocs - name of property, javadoc detail node.
+     * The class javadoc is stored too, with the key being the module name.
+     */
+    private static final Map<String, DetailNode> JAVADOC_FOR_MODULE_OR_PROPERTY =
+            new LinkedHashMap<>();
+
+    /** List of setter methods that the scraper ignores but shouldn't. */
+    private static final Set<String> SUPERCLASS_SETTERS = Set.of(
+        "AbstractFileSetCheck.setFileExtensions",
+        "AbstractHeaderCheck.setHeader"
+    );
+
+    /** Name of the module being scraped. */
+    private static String moduleName = "";
+
+    /**
+     * Initialize the scraper. Clears static context and sets the module name.
+     *
+     * @param newModuleName the module name.
+     */
+    public static void initialize(String newModuleName) {
+        clear();
+        moduleName = newModuleName;
+    }
+
+    /** Clear all scraped information. */
+    private static void clear() {
+        JAVADOC_FOR_MODULE_OR_PROPERTY.clear();
+        moduleName = null;
+    }
+
```

Comment:
This method seems pointless now, we should probably just clear the map in `initialize`, but it is not a big deal.